assimilation_code/modules/utilities/utilities_mod.f90

```diff
@@ -73,6 +73,7 @@ module utilities_mod
           register_module, &
           find_namelist_in_file, &
           check_namelist_read, &
+          flex_parser, &
```

Comment:
i get the intent, but "flex_xx" isn't very specific about what it's flexible about.  it's less catchy, but maybe 'int_from_string_value()' would make the code read more clearly?

int_option = int_from_string_value(string, ...)

seems pretty self-documenting?

Reply:
i still think this is not a good name for this routine.  flex doesn't say what it does, and technically it's not a parser - there is no syntax involved (like "a = b", "bob = call fred()", etc).  i wouldn't bother if it was a module private routine, but this is going into the utilities module for anyone to use and it's just going to propagate a confusing name into more code.

to be clear - i'm fine with the code itself, just not the name of the routine.

Reply:
someone else is going to have to make the call on this.